packages/gatsby-theme-carbon/src/templates/Default.js

```diff
@@ -60,12 +60,14 @@ const Default = ({ pageContext, children, location, Title }) => {
       pageDescription={description}
       pageKeywords={keywords}
       titleType={titleType}>
-      <PageHeader
-        title={Title ? <Title /> : title}
-        label="label"
-        tabs={tabs}
-        theme={theme}
-      />
+      {title && (
+        <PageHeader
+          title={() => <Title />}
```

Comment:
Other than the fix from @vpicone, TIL that you cant just render a `<Title />` as a prop, or maybe this just expects to be a render prop? 

Either way, that is what caused that random error.

Reply:
@barlock Not rendering a page title here isn't desirable as it serves as the `h1` for the page. It's written this way to support the default case (string) as well as a [custom `Title` component](https://gatsby-theme-carbon.now.sh/guides/MDX#custom-title)

Reply:
Ah ha, I wasn't aware of the custom title case.

I'm looking to, if there is no string, and no custom title component, don't render anything. 

I'll update to match that logic.

Reply:
@barlock I thought maybe the plan was to somehow infer the title from the file name? Leaving out the h1 all together is bad news a11y wise. If this is to avoid the ugly error maybe we can improve the error message rather than failing silently with a11y issues?

Reply:
> I thought maybe the plan was to somehow infer the title from the file name? 

I tried this, what's interesting is my attempt at extending `onCreatePage` allowed a `develop` run to build, so the webserver would eventually start, but the rendered page still had an empty `title` (so the page header would render, but be empty). So for some reason this got passed the `NextPrevious` error, but didn't render even ugly.

> Leaving out the h1 all together is bad news a11y wise. 

Agreed, we don't want bad a11y. For pages that don't have a `title` frontmatter, but do have in their markdown a `H1`, we still have an accessible page. 

☝️ Checking this case is possible of course, but it gets complicated and would likely require a custom remark plugin (or something to check the AST). "Trust the developer" on this one seemed like an ok compromise to me.

Reply:
@barlock that's a fair point, although I don't think we have any styles for `h1` on the page, leaving that space empty feels like a mistake.

Reply:
They are indeed there, header is a bit big for my taste, but I think that's just carbon:

![image](https://user-images.githubusercontent.com/5770007/100016052-d8336500-2da6-11eb-8498-3f43dfe36c0c.png)

---

I think I found the bug in my `gatsby-node.js` that will allow me to actually inject it. So, looks like we can calculate it. So, we have options anyway.

I can think of (assumes no title is provided):
1. Default to the base name of the file without the extension (as previously discussed, tis a bit ugly, but works, would end up with two h1s if an h1 is provided)
2. Parse the markdown, search for an H1, inject that, remove from content. 
3. Parse markdown, inject h1 IFF it's the first element, remove the header from content, fail otherwise
4. Parse markdown, search for an H1, if it can't find one, fail, but don't inject it, components don't assume a `title`, (basically don't assume a title, but check for good a11y)
5. Fail in `Default.js` with a better error message if `title === null`

---

My preference would be, in order: 4, 5, 1, 3, 2

Reply:
_hmmm_, Maybe I'm coming back around to 1

Reply:
@barlock yeah I agree, I think one is best. Anything to do with parsing feels a bit fragile